qualtran/bloqs/rotations/hamming_weight_phasing.py

```diff
@@ -74,24 +78,77 @@ class HammingWeightPhasing(GateWithRegisters):
     bitsize: int
     exponent: float = 1
     eps: SymbolicFloat = 1e-10
+    phase_oracle: QvrInterface = attrs.field()
+
+    @phase_oracle.default
+    def _default_qvr(self):
+        return QvrZPow(
+            Register('out', QFxp(self._hamming_weight_bitsize, 0)),
+            gamma=self.exponent / 2,
+            eps=self.eps,
+        )
 
     @cached_property
     def signature(self) -> 'Signature':
-        return Signature.build_from_dtypes(x=QUInt(self.bitsize))
+        return Signature([Register("x", QUInt(self.bitsize)), *self.phase_oracle.extra_registers])
+
+    @classmethod
+    def via_phase_gradient(cls, bitsize: int, exponent: float = 1, eps: SymbolicFloat = 1e-10):
+        r"""Factory method for `HammingWeightPhasing` using a phase gradient state.
+
+        In this variant of Hamming Weight Phasing, instead of directly synthesizing $O(\log_2 (\texttt{bitsize}))$
```

Comment:
I know you copied this, but can you hard-wrap the text to <100 columns. vim is `gq` ; pycharm is "edit -> fill paragraph"